chain/actors/builtin/miner/actor.go.template

```diff
@@ -319,3 +332,37 @@ func AllCodes() []cid.Cid {
     {{- end}}
     }
 }
+
+func PledgePenaltyForContinuedFault(
+		nwVer network.Version,
+		rewardEstimate builtin.FilterEstimate,
+		networkQaPowerEstimate builtin.FilterEstimate,
+		qaSectorPower abi.StoragePower,
+) (abi.TokenAmount, error) {
+	v, err := actorstypes.VersionForNetwork(nwVer)
+	if err != nil {
+		return big.Zero(), err
+	}
+	switch v {
+	{{- range .versions}}
+	case actorstypes.Version{{.}}:
+		{{- if (ge . 16)}}
+			return minertypes{{.}}.PledgePenaltyForContinuedFault(
+				smoothing{{.}}.FilterEstimate{
+					PositionEstimate: rewardEstimate.PositionEstimate,
+					VelocityEstimate: rewardEstimate.VelocityEstimate,
+				},
+				smoothing{{.}}.FilterEstimate{
+					PositionEstimate: networkQaPowerEstimate.PositionEstimate,
+					VelocityEstimate: networkQaPowerEstimate.VelocityEstimate,
+				},
+				qaSectorPower,
+			), nil
+		{{- else}}
+				return big.Zero(), nil
```

Comment:
This is a lie.

Reply:
Honestly, the pledge penalty is the same for all older versions. We could just call into v16 for all versions.

Reply:
great, cause I didn't want to backport the function to all the previous versions